backend/v3/domain/id_provider.go

```diff
@@ -296,6 +296,8 @@ type idProviderColumns interface {
 }
 
 type idProviderConditions interface {
+	// PrimaryKeyCondition returns a filter on the primary key fields.
+	PrimaryKeyCondition(instanceID, idpID string) database.Condition
 	InstanceIDCondition(id string) database.Condition
```

Comment:
Does it still make sense to have this?

I see it is being used internally by `PrimaryKeyCondition()`. Should be removed from the interface. Unless you see some value of having it exported in the interface.